### PR TITLE
Added missing Java ops wrappers

### PR DESCRIPTION
**Pull request by alexanderst**
1. Added some missing wrappers.
2. Added timeouts for tests.

### CODE REVIEW

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    public String opName() {
+        return "random_poisson";
+    }
+
+    @Override
+    public String[] tensorflowNames() {
+        return new String[]{"RandomPoisson","RandomPoissonV2"};
+    }
+
+    private DataType outputDataType = DataType.FLOAT;
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java]: Field at the top of the class

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Override
+    public void initFromTensorFlow(NodeDef nodeDef, SameDiff initWith, Map<String, AttrValue> attributesForNode, GraphDef graph) {
+        if(attributesForNode.containsKey("rate")) {
+            outputDataType = DataTypeAdapter.dtypeConv(attributesForNode.get("rate").getType());
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java]: Shouldn't this be "dtype" not rate?
Rate is the name of the input argument, which has the type attribute "dtype"...

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        }
+    }
+
+    @Override
+    public List<DataType> calculateOutputDataTypes(List<DataType> inputDataTypes){
+        Preconditions.checkState(inputDataTypes != null, "Expected exactly input datatypes for %s, got null", getClass());
+        return Collections.singletonList(outputDataType);
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java:77]: Validation message is incomplete - "expected exactly input datatypes"

**Review by AlexDBlack — COMMENTED** [review]
So far so good. Couple of minor changes required, otherwise looks fine :+1:

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java (hunk 3 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java:77]: Fixed.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java (hunk 2 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java]: Fixed.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java (hunk 1 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java]: Fixed.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomGamma.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
+package org.nd4j.linalg.api.ops.random.custom;
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomGamma.java:17]: Copyright header.
Also for Poisson and Shuffle ops.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java (hunk 3 above)
Comment (AlexDBlack) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java:77]: Still not right. 
```
Preconditions.checkState(inputDataTypes != null && inputDataTypes.size() == 2, "Expected exactly 2 input datatypes for %s, got null", getClass(), inputDataTypes);
```

**Review by AlexDBlack — CHANGES_REQUESTED** [review]

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomGamma.java (hunk 4 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomGamma.java:17]: Fixed.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java (hunk 3 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java:77]: Fixed.

File: nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/custom/CustomOpsTests.java [hunk 5]
```diff
@@ -1353,4 +1351,72 @@ public void testCreateOp_1() {
...
+                9.68627451e-01f, 1.95272727e+02f, 6.11111111e-01f, 7.05882353e-01f}).reshape(5,4,3);
+        INDArray input = Nd4j.createFromArray(new float[]{213.f, 220.f, 164.f, 121.f, 180.f, 180.f,  18.f, 245.f,  75.f, 235.f,  76.f,  74.f, 168.f,
+                50.f, 233.f, 191.f, 132.f, 100.f, 207.f,  37.f, 245.f,  77.f, 250.f, 182.f, 111.f,  52.f,
+                59.f, 193.f, 147.f, 137.f, 168.f, 103.f, 121.f,  48.f, 191.f, 187.f,  53.f,  82.f, 239.f,
+                156.f,  37.f, 118.f, 244.f,  90.f,   7.f, 221.f,  98.f, 243.f,  12.f, 209.f, 192.f,   2.f,
+                115.f, 205.f,  79.f, 247.f,  32.f,  70.f, 152.f, 180.f}).reshape(5,4,3);
+        RgbToHsv op = new RgbToHsv(input);
+        INDArray[] ret = Nd4j.exec(op);
+        assertEquals(ret[0], expected);
+    }
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/custom/CustomOpsTests.java:1381]: For these sorts of tests, can you include info about how to reproduce them?
For example a few lines of python as a comment assuming you pulled it out of TF.
The problem with these sorts of tests is that if they fail, we have no idea if it's a bad test, some edge case, or an actual op failure.

Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/custom/CustomOpsTests.java:1381]: Added small comments.

**Review by AlexDBlack — APPROVED** [review]
LGTM :+1:
